gdal/ogr/ogrsf_frmts/gml/ogrgmldatasource.cpp

```diff
@@ -873,12 +873,13 @@ bool OGRGMLDataSource::Open( GDALOpenInfo *poOpenInfo )
                     GMLRegistryNamespace &oNamespace =
                         oRegistry.aoNamespaces[iNS];
                     const char *pszNSToFind =
-                        CPLSPrintf("xmlns:%s", oNamespace.osPrefix.c_str());
+                        (oNamespace.osPrefix.size()) ? CPLSPrintf("xmlns:%s", oNamespace.osPrefix.c_str()) : NULL;
```

Comment:
!oNamespace.osPrefix.empty() would be more consistant with what is done elsewhere in the code